target/linux/ath79/dts/qca9531_compex_wpj531.dts

```diff
@@ -0,0 +1,131 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "qca953x.dtsi"
+
+/ {
+	compatible = "compex,wpj531", "qca,qca9531";
+	model = "Compex WPJ531 (16MB flash)";
+
+	chosen {
+		bootargs = "console=ttyS0,115200n8";
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		sig1 {
+			label = "wpj531:red:sig1";
+			gpios = <&gpio 12 GPIO_ACTIVE_LOW>;
+		};
+
+		sig2 {
+			label = "wpj531:yellow:sig2";
+			gpios = <&gpio 14 GPIO_ACTIVE_LOW>;
+		};
+
+		sig3 {
+			label = "wpj531:green:sig3";
+			gpios = <&gpio 15 GPIO_ACTIVE_LOW>;
+		};
+
+		sig4 {
+			label = "wpj531:green:sig4";
+			gpios = <&gpio 16 GPIO_ACTIVE_LOW>;
+		};
```

Comment:
What are those leds for? You do not seem to use them btw...

Reply:
compex calls the LEDs `RSSI1` to `RSSI4`.
afaik, openwrt doesn't have any infrastructure for such a function.
i'm happy with the compromise of calling them `sig` in ar71xx and the compex SDK.

other boards have sections like this:
```
  aliases {
    led-boot = &qss_led;
    led-failsafe = &qss_led;
    led-running = &qss_led;
    led-upgrade = &qss_led;
  };
```
i don't understand how it is implemented but `RSSI4/Diag` could become a general status indicator.

otherwise, the LED definitions are mainly there to create corresponding files in `/sys/class/led`.

Reply:
aliases:
You can make one of the leds a system led by giving it a name, e.g.
`system: sig4 {` instead of `sig4 {`.

Then you could reference it in aliases:
```
aliases {
    led-boot = &system;
    led-failsafe = &system;
    led-running = &system;
    led-upgrade = &system;
  };
```

This LED will then blink on boot etc.

My question was aiming into a different direction:

> otherwise, the LED definitions are mainly there to create corresponding files in /sys/class/led

So, you plan to define them, so other people can use them, but do not imply a default use?

Reply:
thank you for confirming the functionality of the aliases.
sounds promising.
if there's an openwrt-way of using the LEDs, i think we should adopt it.
otherwise, there is no consistent in behaviour in the firmwares supplied by compex.

there are 7 leds on the board:
- power
- eth0
- eth1
- rssi1
- rssi2
- rssi3
- rssi4/diag

if i recall correctly a discussion with compex staff, the RSSI LEDs correspond to MIMO streams on both bands in compexWRT - but not in the SDK (where they are called sig1-4).

> So, you plan to define them, so other people can use them, but do not imply a default use?
yes. but that's mainly because i like to use the sig4/diag in applications.
and i don't see why they shouldn't be included if openwrt knows how to drive them.

Reply:
So, can we implement eth0, eth1 and power? (Or have I overlooked them?)

Reply:
the power led is done in hardware.
for some reason, i assumed that would also be the case for eth0 and eth1 - it isn't.
i'll add that and sysupgrade compatibility for ar71xx.

it'll be a while before i can get the untouched boards out of storage.